lib/ansible/module_utils/common/parameters.py

```diff
@@ -0,0 +1,124 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2018 Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or https://opensource.org/licenses/BSD-2-Clause)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+from ansible.module_utils._text import to_native
+from ansible.module_utils.common._collections_compat import Mapping
+from ansible.module_utils.common.collections import is_iterable
+
+from ansible.module_utils.six import (
+    binary_type,
+    integer_types,
+    text_type,
+)
+
+# Python2 & 3 way to get NoneType
+NoneType = type(None)
+
+PASS_VARS = {
+    'check_mode': 'check_mode',
+    'debug': '_debug',
+    'diff': '_diff',
+    'keep_remote_files': '_keep_remote_files',
+    'module_name': '_name',
+    'no_log': 'no_log',
+    'remote_tmp': '_remote_tmp',
+    'selinux_special_fs': '_selinux_special_fs',
+    'shell_executable': '_shell',
+    'socket': '_socket_path',
+    'syslog_facility': '_syslog_facility',
+    'tmpdir': '_tmpdir',
+    'verbosity': '_verbosity',
+    'version': 'ansible_version',
+}
+
+DEFAULT_LEGAL_PARAMS = ['_ansible_%s' % k for k in PASS_VARS]
+
+
+def handle_aliases(argument_spec, params, legal_inputs=None):
+    """Return a two items. The first is a dictionary of aliases, the second is
+    a list of legal inputs, or None."""
+
+    if legal_inputs is None:
+        legal_inputs = DEFAULT_LEGAL_PARAMS[:]
```

Comment:
Looking at this, I wonder if maybe we should just do this here, and start out `self._legal_inputs` from `AnsibleModule.__init__` as an empty list, and then there would be no reason to pass in `legal_inputs` to begin with.  Just make this function responsible for setting it, since that is the end result anyway.